File: leetcode/594 Longest Harmonious Subsequence/solution.go
func findLHS(nums []int) int {
    hashRecord := make(map[int]int)
    for _, num := range nums {
        hashRecord[num]++
    }

    result := 0
    for key, count := range hashRecord {
        if hashRecord[key+1] != 0 {
            b := hashRecord[key+1]
            result = max(result, count+b)
        }
    }

    return result
}

func max(a int, b int) int {
    if a >= b {
        return a
    } else {
        return b
    }
}

File: leetcode/594 Longest Harmonious Subsequence/solution.java
class Solution {
    public int findLHS(int[] nums) {
        int n = nums.length;
        Map <Integer, Integer> hashRecord = new HashMap<>();
        for (int num : nums) {
            hashRecord.put(num, hashRecord.getOrDefault(num, 0) + 1);
        }
 
        int result = 0;
        for (int key : hashRecord.keySet()) {
            int a = hashRecord.get(key);
            if (hashRecord.containsKey(key+1)) {
                int b = hashRecord.get(key+1);
                result = Math.max(result, a+b);
            }           
        }

        return result;
    }
}